### ISSUE TITLE
Nearpc command fails with gdb.error: Cannot convert value to long

### ISSUE THREAD

**Issue by tmfink** [pwndbg/pwndbg#120]
I ran into this bug when using pwndbg with the pwnable.kr problem `brainf`:

bf mirror: https://dl.dropboxusercontent.com/u/2564904/bf

Running `gdb -x nearpc_test.gdb ./bf`

Where nearpc_test.gdb is:

```
show configuration
b main
r
nearpc main
```

Yields the following in Ubuntu 16.04: 

```
$ gdb -x nearpc_test.gdb ./bf
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.04) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./bf...(no debugging symbols found)...done.
Loaded 94 commands.  Type pwndbg [filter] for a list.
This GDB was configured as follows:
   configure --host=x86_64-linux-gnu --target=x86_64-linux-gnu
             --with-auto-load-dir=$debugdir:$datadir/auto-load
             --with-auto-load-safe-path=$debugdir:$datadir/auto-load
             --with-expat
             --with-gdb-datadir=/usr/share/gdb (relocatable)
             --with-jit-reader-dir=/usr/lib/gdb (relocatable)
             --without-libunwind-ia64
             --with-lzma
             --with-python=/usr (relocatable)
             --without-guile
             --with-separate-debug-dir=/usr/lib/debug (relocatable)
             --with-system-gdbinit=/etc/gdb/gdbinit
             --with-babeltrace

("Relocatable" means the directory can be moved with the GDB installation
tree, and GDB will still find it.)
Breakpoint 1 at 0x8048675

Breakpoint 1, 0x08048675 in main ()
Traceback (most recent call last):
  File "/home/ryucl0ud/workspace/pwndbg/pwndbg/commands/__init__.py", line 54, in __call__
    return self.function(*args, **kwargs)
  File "/home/ryucl0ud/workspace/pwndbg/pwndbg/commands/__init__.py", line 112, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/home/ryucl0ud/workspace/pwndbg/pwndbg/commands/nearpc.py", line 44, in nearpc
    if lines is None and (pc is None or int(pc) < 0x100):
gdb.error: Cannot convert value to long.

LEGEND: STACK | HEAP | CODE | DATA | RWX | RODATA
[───────────────────────────────────REGISTERS────────────────────────────────────]
*EAX  0xf7fa8ddc (environ) —▸ 0xffffd61c —▸ 0xffffd7f8 ◂— 'XDG_VTNR=7'
 EBX  0x0
*ECX  0xa298b894
*EDX  0xffffd5a4 ◂— 0x0
*EDI  0xf7fa7000 (_GLOBAL_OFFSET_TABLE_) ◂— 0x1b1db0
*ESI  0xf7fa7000 (_GLOBAL_OFFSET_TABLE_) ◂— 0x1b1db0
*EBP  0xffffd578 ◂— 0x0
*ESP  0xffffd574 ◂— 0x0
*EIP  0x8048675 (main+4) ◂— and    esp, 0xfffffff0
[──────────────────────────────────────CODE──────────────────────────────────────]
 ► 0x8048675 <main+4>     and    esp, 0xfffffff0
   0x8048678 <main+7>     sub    esp, 0x430
   0x804867e <main+13>    mov    eax, dword ptr [ebp + 0xc]
   0x8048681 <main+16>    mov    dword ptr [esp + 0x1c], eax
   0x8048685 <main+20>    mov    eax, dword ptr gs:[0x14]
   0x804868b <main+26>    mov    dword ptr [esp + 0x42c], eax
   0x8048692 <main+33>    xor    eax, eax
   0x8048694 <main+35>    mov    eax, dword ptr [stdout@@GLIBC_2.0] <0x804a060>
   0x8048699 <main+40>    mov    dword ptr [esp + 0xc], 0
   0x80486a1 <main+48>    mov    dword ptr [esp + 8], 2
   0x80486a9 <main+56>    mov    dword ptr [esp + 4], 0
[─────────────────────────────────────STACK──────────────────────────────────────]
00:0000│ esp  0xffffd574 ◂— 0x0
... ↓
02:0008│      0xffffd57c —▸ 0xf7e0d637 (__libc_start_main+247) ◂— add    esp, 0x10
03:000c│      0xffffd580 ◂— 0x1
04:0010│      0xffffd584 —▸ 0xffffd614 —▸ 0xffffd7c3 ◂— '/home/...'
05:0014│      0xffffd588 —▸ 0xffffd61c —▸ 0xffffd7f8 ◂— 'XDG_VTNR=7'
06:0018│      0xffffd58c ◂— 0x0
... ↓
[───────────────────────────────────BACKTRACE────────────────────────────────────]
 ► f 0  8048675 main+4
   f 1 f7e0d637 __libc_start_main+247
Breakpoint main
pwndbg> 
```

**Comment by gsingh93** [comment]
This is happening because the type of `main` is `gdb.TYPE_CODE_FUNC`, and calling `int` on that doesn't work. I'm wondering if we should add something to [inthook.py](https://github.com/pwndbg/pwndbg/blob/24c4847eebe964f78f5885241f47d5bfe5fff899/pwndbg/inthook.py) to generically handle this case.

**Comment by qxxxb** [comment]
Not sure why, but I'm getting the same error again, and it seems to happen on any executable file and any function name.

pwndbg: dev branch
GDB: 10.1
OS: OS: Ubuntu 18.04.5 LTS x86_64
Terminal: kitty

```
 gdb passcode
GNU gdb (GDB) 10.1
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
pwndbg: loaded 193 commands. Type pwndbg [filter] for a list.
pwndbg: created $rebase, $ida gdb functions (can be used with print/break)
Reading symbols from passcode...
(No debugging symbols found in passcode)
pwndbg> start
Temporary breakpoint 1 at 0x80484b0
pwndbg> 
Temporary breakpoint 1, 0x080484b0 in _start ()
set exception-verbose on
Set whether to print a full stacktrace for exceptions raised in Pwndbg commands to True
LEGEND: STACK | HEAP | CODE | DATA | RWX | RODATA
────────────────────────────────────────────────────────────────────────────────────────[ REGISTERS ]────────────────────────────────────────────────────────────────────────────────────────
*EAX  0xf7ffd940 ◂— 0
*EBX  0xf7ffd000 (_GLOBAL_OFFSET_TABLE_) ◂— xor    al, 0x6f /* 0x26f34 */
*ECX  0x0
*EDX  0xf7fe59e0 (_dl_fini) ◂— push   ebp /* 0x57e58955 */
*EDI  0x80484b0 (_start) ◂— xor    ebp, ebp
*ESI  0xffffc37c —▸ 0xffffc5d2 ◂— 'USER=plushie'
*EBP  0x0
*ESP  0xffffc370 ◂— 0x1
*EIP  0x80484b0 (_start) ◂— xor    ebp, ebp
─────────────────────────────────────────────────────────────────────────────────────────[ DISASM ]──────────────────────────────────────────────────────────────────────────────────────────
 ► 0x80484b0 <_start>       xor    ebp, ebp
   0x80484b2 <_start+2>     pop    esi
   0x80484b3 <_start+3>     mov    ecx, esp
   0x80484b5 <_start+5>     and    esp, 0xfffffff0
   0x80484b8 <_start+8>     push   eax
   0x80484b9 <_start+9>     push   esp
   0x80484ba <_start+10>    push   edx
   0x80484bb <_start+11>    push   __libc_csu_fini <0x8048710>
   0x80484c0 <_start+16>    push   __libc_csu_init <0x80486a0>
   0x80484c5 <_start+21>    push   ecx
   0x80484c6 <_start+22>    push   esi
──────────────────────────────────────────────────────────────────────────────────────────[ STACK ]──────────────────────────────────────────────────────────────────────────────────────────
00:0000│ esp  0xffffc370 ◂— 0x1
01:0004│      0xffffc374 —▸ 0xffffc59a ◂— '/home/plushie/Programs/ctf/pwnable_kr/passcode/passcode'
02:0008│      0xffffc378 ◂— 0x0
03:000c│ esi  0xffffc37c —▸ 0xffffc5d2 ◂— 'USER=plushie'
04:0010│      0xffffc380 —▸ 0xffffc5df ◂— 'LANGUAGE=en_US'
05:0014│      0xffffc384 —▸ 0xffffc5ee ◂— 'XDG_SEAT=seat0'
06:0018│      0xffffc388 —▸ 0xffffc5fd ◂— 'XDG_SESSION_TYPE=x11'
07:001c│      0xffffc38c —▸ 0xffffc612 ◂— 'SSH_AGENT_PID=1343'
────────────────────────────────────────────────────────────────────────────────────────[ BACKTRACE ]────────────────────────────────────────────────────────────────────────────────────────
 ► f 0  80484b0 _start
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
pwndbg> nearpc main
Traceback (most recent call last):
  File "/home/plushie/Programs/archive/pwndbg/pwndbg/commands/__init__.py", line 130, in __call__
    return self.function(*args, **kwargs)
  File "/home/plushie/Programs/archive/pwndbg/pwndbg/commands/__init__.py", line 221, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/home/plushie/Programs/archive/pwndbg/pwndbg/commands/nearpc.py", line 61, in nearpc
    if lines is None and (pc is None or int(pc) < 0x100):
gdb.error: Cannot convert value to long.

If that is an issue, you can report it on https://github.com/pwndbg/pwndbg/issues
(Please don't forget to search if it hasn't been reported before)
To generate the report and open a browser, you may run `bugreport --run-browser`
PS: Pull requests are welcome
pwndbg> 
```

**Comment by disconnect3d** [comment]
As a workaround, use `nearpc &main`

**Comment by disconnect3d** [comment]
Finally fixed in #1048 thx to @l4rzy ! :D
